x-pack/plugins/observability/public/hooks/use_guided_setup.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useState, useCallback } from 'react';
+
+export const LOCAL_STORAGE_HIDE_GUIDED_SETUP_KEY = 'HIDE_GUIDED_SETUP';
+
+export function useGuidedSetup() {
+  const hideGuidedSetupLocalStorageKey = window.localStorage.getItem(
+    LOCAL_STORAGE_HIDE_GUIDED_SETUP_KEY
+  );
+  const [isGuidedSetupHidden, setIsGuidedSetupHidden] = useState<boolean>(
+    JSON.parse(hideGuidedSetupLocalStorageKey || 'false')
+  );
+
+  const hideGuidedSetup = useCallback(() => {
```

Comment:
We don't need a "show" method at the moment?

Reply:
No, this is to hide the progress bar at the top of the page.

Maybe `dismissGuidedSetup` is more clear?

Reply:
Ohhh ok thanks. I like `dismiss` for that, yeah!

Reply:
I think it just makes me think it's hiding/dismissing the actual Guided Setup view which, to me, seems like the flyout with all of the instructions in it. What do we call the flyout and what do we call the progress bar, in code?

Reply:
> What do we call the flyout and what do we call the progress bar, in code?

`<ObservabilityStatus />` and `<ObservabilityStatusProgress />`. I think that was the name we gave them in the beginning and that's what have stayed. 

I think moving forward we want to use "Guided Setup" everywhere. For now I'll rename the callback to `dismissGuidedSetupProgress` and address terminology later.